client/src/controllers/MessagesController.ts

```diff
@@ -35,56 +23,99 @@ export class MessagesController extends Controller<HTMLElement> {
-    }
+  static afterLoad() {
+    window.handleUnsavedEvent = (event) => {
+      const { formDirty, commentsDirty } = event.detail;
+      const anyDirty = formDirty || commentsDirty;
+      const allDirty = formDirty && commentsDirty;
+      const type = allDirty ? "all" : commentsDirty ? "comments" : "edits";
+      const detail = { type };
+      const eventName = anyDirty ? "w-unsaved:show" : "w-unsaved:hide";
+      document.dispatchEvent(new CustomEvent(eventName, { detail }));
+    };
+  }
 
-    if (clear) this.clear();
+add(
+  event?: CustomEvent<{
+    /** Flag for clearing or stacking messages */
+    clear?: boolean;
+    /** Content for the message, HTML not supported. */
+    text?: string;
+    /** Message status level, based on Django's message types. */
+    type?: 'success' | 'error' | 'warning' | string;
+  }>
+) {
+  const { clear = false, text = '', type } = event?.detail || {};
 
-    /** if no type provided, return the first template target, otherwise try to find
-     * a matching target, finally fall back on the first template target if nothing
-     * is found.
-     */
-    const template =
-      (type &&
-        this.templateTargets.find(({ dataset }) => dataset.type === type)) ||
-      this.templateTarget;
+  if (this.hasAddedClass) {
+    this.element!.classList.add(this.addedClass);
+  }
+
+  if (clear) {
+    setTimeout(() => {
+      this.clear().then(() => {
+        this.add(event);
+      });
+    }, this.clearDelayValue);
+    return;
+  }
 
-    const content = template.content.firstElementChild?.cloneNode(true);
+  const template =
+    (type &&
+      this.templateTargets.find(({ dataset }) => dataset.type === type)) ||
+    this.templateTarget;
 
-    if (content instanceof HTMLElement) {
-      const textElement = content.lastElementChild;
+  const content = template.content.firstElementChild?.cloneNode(true);
 
-      if (textElement instanceof HTMLElement && text) {
-        textElement.textContent = text;
-      }
+  if (content instanceof HTMLElement) {
+    const textElement = content.lastElementChild;
+
+    if (textElement instanceof HTMLElement && text) {
+      textElement.textContent = text;
+    }
 
-      this.containerTarget.appendChild(content);
+    this.containerTarget.appendChild(content);
 
-      this.dispatch('added');
+    this.dispatch('added');
 
-      if (this.hasShowClass) {
-        setTimeout(() => {
-          this.element.classList.add(this.showClass);
-        }, this.showDelayValue);
-      }
+    if (this.hasShowClass) {
+      setTimeout(() => {
+        this.element.classList.add(this.showClass);
+      }, this.showDelayValue);
     }
   }
+}
+
 
   clear() {
-    this.containerTarget.innerHTML = '';
+    return new Promise((resolve) => {
+      this.containerTarget.classList.remove("footer__container--visible");
+      setTimeout(() => {
+        this.containerTarget.innerHTML = "";
+        this.containerTarget.hidden = true;
+        resolve(undefined);
```

Comment:
```suggestion
        resolve();
```

I do not think you need to pass in a value here.